src/sonic-yang-models/yang-models/sonic-evpn.yang

```diff
@@ -0,0 +1,146 @@
+module sonic-evpn {
+    yang-version 1.1;
+
+    namespace "http://github.com/sonic-net/sonic-evpn";
+    prefix sevpn;
+
+    import sonic-interface {
+        prefix sinterface;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC EVPN";
+
+    revision 2024-05-08 {
+        description
+            "First revision.";
+    }
+
+    revision 2024-06-14 {
+        description
+            "Add global EVPN Multihoming configuration.";
+    }
+
+    typedef esi-type {
+        type enumeration {
+            enum TYPE_0_OPERATOR_CONFIGURED {
+                value 0;
+                description "Type 0 ESI - Operator Configured";
+            }
+            enum TYPE_3_MAC_BASED {
+                value 3;
+                description "Type 3 ESI - MAC-Based ESI";
+            }
+        }
+    }
+
+    container sonic-evpn {
+
+        container EVPN_ETHERNET_SEGMENT {
+            description "CONFIG_DB EVPN_ETHERNET_SEGMENT Table";
+
+            list EVPN_ETHERNET_SEGMENT_LIST {
+
+                key "ifname";
+
+                /*
+                 * It's not possible to have a leafref pointing to two separate models
```

Comment:
https://github.com/sonic-net/sonic-buildimage/blob/6f110acf2b9cd18c93ada95cae24ea367791bcb9/src/sonic-yang-models/yang-models/sonic-system-tacacs.yang#L149-L156

how about union to point to both?

Reply:
Thank you for the suggestion. I have made this chamge.